src/node_modules/logger.js

```diff
@@ -395,58 +393,60 @@ export default class Logger extends Writable {
 		});
 
 		this[$connection] = new Promise((resolve, reject) => {
-			const debugLogger = this.debugLogger;
 			const connOpts = {
 				host: this.host,
 				port: this.port
-			};			
+			};
 
-			this[$reconnection].on('connect', function (con) {
-				debugLogger.log("Connected");
+			this[$reconnection].on('connect', (connection) => {
+				this.debugLogger.log("Connected");
 				this.emit('connected');
```

Comment:
yep, arrow function expression makes sure of that: https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/Arrow_functions